src/scripts/import-needs-assessment-data/add-regions.ts

```diff
@@ -0,0 +1,241 @@
+  return value.status === "rejected";
+};
+
+/*  Consolidate Regions
+ * ------------------------------------------------------- */
+function consolidateRegions(data: NeedAssessment[]): string[] {
+  const regions = data.reduce((acc: string[], item) => {
+    const region = item.place?.region;
+    if (region && !acc.includes(region)) {
+      acc.push(region);
+    }
+    return acc;
+  }, []);
+
+  return regions;
+}
+
+/*  Parse Region
+ * ------------------------------------------------------- */
+function parseRegion({
+  data,
+  orig,
+  status,
+  logs,
+}: RegionUploadWorkflow): RegionUploadWorkflow {
+  logs = [...logs, `Log: parsing region "${orig}"`];
+
+  if (orig == null || typeof orig !== "string") {
+    throw {
+      data,
+      orig,
+      status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+      logs: [
+        ...logs,
+        `Error: Invalid region input: "${orig}". Expected a non-null value.`,
+      ],
+    };
+  }
+
+  const newData = {
+    ...data,
+    region: orig,
+  };
+
+  return {
+    data: newData,
+    orig,
+    status,
+    logs,
+  };
+}
+
+/*  Get Region
+ * ------------------------------------------------------- */
+async function getRegion({
+  data,
+  orig,
+  status,
+  logs,
+}: RegionUploadWorkflow): Promise<RegionUploadWorkflow> {
+  logs = [...logs, `Log: Checking if Geo.Region "${orig}" already exists.`];
+
+  //Fetch the data from Strapi
+  const response = await fetch(`${STRAPI_ENV.URL}/regions?`, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${STRAPI_ENV.KEY}`,
+    },
+  });
+
+  const body = await response.json();
+  const matchingRegion = body.data.find(
+    (region) => region.Name.toLowerCase() === data.region.toLowerCase(),
+  );
+
+  if (!response.ok) {
+    console.log("Non-ok response");
+    throw {
+      data,
+      orig,
+      status: UploadWorkflowStatus.DUPLICATE_CHECK_ERROR,
+      logs: [
+        ...logs,
+        `Error: Failed to get Geo.Region. HttpStatus: ${response.status} - ${response.statusText}`,
+        JSON.stringify(body),
+      ],
+    };
+  }
+
+  if (matchingRegion) {
+    throw {
+      data: body.data,
+      orig,
+      status: UploadWorkflowStatus.ALREADY_EXISTS,
+      logs: [...logs, "Log: Found existing Geo.Region. Skipping..."],
+    };
+  }
+
+  return {
+    data,
+    orig,
+    status,
+    logs: [...logs, "Success: Confirmed Geo.Region does not exist."],
+  };
+}
+
+/*  Upload Region
+ * ------------------------------------------------------- */
+async function uploadRegion({
+  data,
+  orig,
+  /* status, */ logs,
+}: RegionUploadWorkflow): Promise<RegionUploadWorkflow> {
+  logs = [...logs, `Log: Creating Geo.Region "${orig}".`];
+
+  const response = await fetch(`${STRAPI_ENV.URL}/regions`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${STRAPI_ENV.KEY}`,
+    },
+    body: JSON.stringify({
+      data: {
```

Comment:
Can we do a validation of the data object before we send it. If we know the know the shape of it ahead of time we could create an UploadRegionType to make sure the payload is correct. Unless we already have one